src/components/image-plus.js

```diff
@@ -0,0 +1,252 @@
+        height = geo.height * ratio;
+      }
+    } else if (geo && geo.height) {
+      width = geo.width / ratio;
+    } else {
+      width = Math.min(1.0, 1.0 / ratio);
+      height = Math.min(1.0, ratio);
+    }
+    this.el.setAttribute("geometry", { width, height });
+    this.el.setAttribute("shape", {
+      shape: "box",
+      halfExtents: {
+        x: width / 2,
+        y: height / 2,
+        z: 0.05
+      }
+    });
+  },
+
+  init() {
+    const material = new THREE.MeshBasicMaterial();
+    material.side = THREE.DoubleSide;
+    material.transparent = true;
+    this.el.getObject3D("mesh").material = material;
+  },
+
+  remove() {
+    const material = this.el.getObject3D("mesh").material;
+    const texture = material.map;
+
+    if (texture === errorTexture) return;
+
+    const url = texture.image.src;
+    const cacheItem = textureCache.get(url);
+    cacheItem.count--;
+    if (cacheItem.count <= 0) {
+      console.log("removing from cache");
+      // Unload the video element to prevent it from continuing to play in the background
+      if (texture.image instanceof HTMLVideoElement) {
+        const video = texture.image;
+        video.pause();
+        video.src = "";
+        video.load();
+      }
+
+      texture.dispose();
+
+      // THREE never lets go of material refs, long running PR HERE https://github.com/mrdoob/three.js/pull/12464
+      // Mitigate the damage a bit by at least breaking the image ref so Image/Video elements can be freed
+      // TODO: If/when THREE gets fixed, we should be able to safely remove this
+      delete texture.image;
+
+      textureCache.delete(url);
+    }
+  },
+
+  async loadGIF(url) {
+    return new Promise((resolve, reject) => {
+      // TODO: pool workers
+      const worker = new GIFWorker();
+      worker.onmessage = e => {
+        const [success, frames, delays, disposals] = e.data;
+        if (!success) {
+          reject(`error loading gif: ${e.data[1]}`);
+          return;
+        }
+
+        let loadCnt = 0;
+        for (let i = 0; i < frames.length; i++) {
+          const img = new Image();
+          img.onload = e => {
+            loadCnt++;
+            frames[i] = e.target;
+            if (loadCnt === frames.length) {
+              const texture = new GIFTexture(frames, delays, disposals);
+              texture.image.src = url;
+              resolve(texture);
+            }
+          };
+          img.src = frames[i];
+        }
+      };
+      fetch(url, { mode: "cors" })
+        .then(r => r.arrayBuffer())
+        .then(rawImageData => {
+          worker.postMessage(rawImageData, [rawImageData]);
+        })
+        .catch(reject);
+    });
+  },
+
+  loadVideo(url) {
+    return new Promise((resolve, reject) => {
+      const videoEl = createVideoEl(url);
+
+      const texture = new THREE.VideoTexture(videoEl);
+      texture.minFilter = THREE.LinearFilter;
+      videoEl.addEventListener("loadedmetadata", () => resolve(texture));
```

Comment:
```
videoEl.addEventListener("loadedmetadata", () => resolve(texture), {once: true});
```

Reply:
woah, that's useful. I didn't know about that `once` flag